src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs

```diff
@@ -100,24 +100,38 @@ public ShellSettings Match(string host, string appRelativePath)
         private bool TryMatchInternal(string host, string appRelativePath, out ShellSettings result)
         {
             // 1. Search for Host + Prefix match
-            string hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
+            var hostAndPrefix = GetHostAndPrefix(host, appRelativePath);
 
             if (!_shellsByHostAndPrefix.TryGetValue(hostAndPrefix, out result))
             {
-                // 2. Search for Host only match
+                // 2. Search for Full Host only match
```

Comment:
What I see here is a mis-configuation of your tenants, you set localhost where actually you should not have any host specified.